src/ol/geom/linestring.js

```diff
@@ -246,7 +246,7 @@ ol.geom.LineString.prototype.intersectsExtent = function(extent) {
  */
 ol.geom.LineString.prototype.setCoordinates = function(coordinates, opt_layout) {
   if (!coordinates) {
-    this.setFlatCoordinates(ol.geom.GeometryLayout.XY, null);
+    this.setFlatCoordinates(ol.geom.GeometryLayout.XY, []);
```

Comment:
I've already started to investigate. Checked with master and flatCoordinates is set to null but when getCoordinates is called it is an empty array. Looking into it.